zio-http/src/test/scala/zhttp/service/ServerIssue1229Spec.scala

```diff
@@ -0,0 +1,63 @@
+package zhttp.service
+
+import zhttp.http._
+import zhttp.service.server.ServerChannelFactory
+import zio._
+import zio.stream.ZStream
+import zio.test.TestAspect.timeout
+import zio.test._
+
+/**
+ * See https://github.com/dream11/zio-http/issues/1229
+ */
+object ServerIssue1229Spec extends ZIOSpecDefault {
+  private def callWithContent(content: HttpData): Task[Response] = {
+    val dummyRoute = Http.collect[Request] { case req @ Method.POST -> !! / "echo" =>
+      Response(status = Status.Ok, data = req.data)
+    }
+
+    val layer: ZLayer[ServerChannelFactory & EventLoopGroup, Nothing, Server.Start] = ZLayer.scoped {
+      Server(dummyRoute).withBinding("localhost", 0).make.orDie
+    }
+    val env                                                                         = ZLayer
+      .make[Server.Start & EventLoopGroup & ChannelFactory](
+        layer,
+        EventLoopGroup.auto(0),
+        ServerChannelFactory.auto,
+        ChannelFactory.auto,
+      )
+
+    val action = for {
+      port <- ZIO.serviceWith[Server.Start](_.port)
+      resp <- Client.request(
+        s"http://localhost:$port/echo",
+        method = Method.POST,
+        content = content,
+      )
+    } yield resp
+
+    action.provideLayer(env)
+  }
+
+  def spec = suite("ServerIssue1229Spec")(
+    test("fromString") {
+      for {
+        resp           <- callWithContent(HttpData.fromString("echo string"))
+        responseString <- resp.bodyAsString
+      } yield assertTrue(resp.status.isSuccess, responseString == "echo string")
+    },
+    test("empty") {
+      for {
+        resp           <- callWithContent(HttpData.empty)
+        responseString <- resp.bodyAsString
+      } yield assertTrue(resp.status.isSuccess, responseString == "")
+    },
+    test("fromStream") {
+      val stream = ZStream.repeatZIOChunk(ZIO.sleep(10.millis).as(Chunk[String]("A"))).take(10)
+      for {
+        resp           <- callWithContent(HttpData.fromStream(stream))
+        responseString <- resp.bodyAsString
+      } yield assertTrue(resp.status.isSuccess, responseString == "AAAAAAAAAA")
+    },
+  ) @@ timeout(5.seconds)
```

Comment:
We can add a simple test in `ServerSpec.scala` where we are just echoing.

Reply:
I've tried to reproduce the initial issue with `ServerSpec` approach with no luck.